functions/Set-DbaMaxMemoryEachInstanceBySize.ps1

```diff
@@ -0,0 +1,177 @@
+If any instance got 2GB, then it adjusts the calculation to accomodate an instance getting more than would have
+been calculated.
+
+Jonathan's max server memory recommenation: 
+https://www.sqlskills.com/blogs/jonathan/how-much-memory-does-my-sql-server-actually-need/
+
+.PARAMETER Server
+Server or collection of servers, not a SQL Server instance but rather hostname
+
+.PARAMETER SqlCredential
+Use SqlCredential to connect to each instance found on Server with SQL authentication. 
+If SqlCredential is not specified, Windows authentication will be used.
+
+Note that as the function looks at services to determine which SQL Server instances are on the server, 
+the account running the function needs admin access to Windows.
+
+.NOTES 
+Original author: Tara Kizer, Brent Ozar Unlimited (https://www.brentozar.com/)
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Set-DbaMaxMemoryeachInstanceBySize
+
+.EXAMPLE
+Set-DbaMaxMemoryeachInstanceBySize -Server server1
+
+Connects to server1, connects to each SQL Server instance on server1 with Windows authentication
+and then sets max server memory for each instance found based on size of databases
+
+.EXAMPLE
+Set-DbaMaxMemoryeachInstanceBySize -Server server1 -SqlCredential $cred
+
+Connects to server1, connects to each SQL Server instance on server1 with SQL authentication
+and then sets max server memory for each instance found based on size of databases
+
+#>
+	
+	[CmdletBinding()]
+	param (
+		[Parameter(Mandatory = $True, ValueFromPipeline = $True)]
+		[object]$Server,
+        [object]$SqlCredential
+	)
+	
+	BEGIN { }
+
+    PROCESS {
+
+		$Services = Get-DbaSqlService -ComputerName $Server -Type Engine | Out-DbaDataTable
+
+		$Instances = @() # list of instances
+		$InstanceSizes = @() # list of instance name, instance size
+		$i = 0
+		$TotalServerSize = 0
+
+		# Get the SQL Server service names, convert them to ServerName or ServerName\InstanceName
+		foreach ($Service in $Services | Where-Object {$_.State -eq "Running"}) {
+			$Instance = $Server + "\" + $Service[1]
+			$Instance = $Instance.replace("\MSSQLSERVER", "").replace("MSSQL$","")
+
+			$Instances += ,@($Instance)
+		}
+
+		foreach ($Instance in $Instances) {
```

Comment:
I think that a check could be skipped since it simply won't be looped since $instances won't exist